### PR TITLE
Project Cleanup Take 2

### PR DESCRIPTION
**Pull request by purinda**
### Tasks
- [x] Remove unwanted ansible configurations
- [x] Pre tasks required for ubuntu distros (python 2.x setup) should be moved from sample.yml to a proper role.
- [x] Checks to make sure `awscli` is properly setup under localhost
- [x] OSX should run pip installations for user (context)
- [x] pip package installations should run with `--user` for OSX and as `sudo` for Linux
- [x] README.md should be improved with examples, FAQs and known issues etc.

### Fixes
- Security group, outgoing rules indentation fix.

### Tests
- [x] Check with Amazon AMIs
- [x] Check with Ubuntu 16.04

### CODE REVIEW

File: src/os.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[ $CLI_OS ] && return || CLI_OS=1
```
Comment (jordonsc) [inline src/os.sh]: Convention would have this be `$OS_SH`

File: ansible/roles/docker/tasks/main.yml [hunk 2]
```diff
@@ -63,9 +63,11 @@
     - PyYAML
 
 - name: Enable docker
-  service: 
-    enabled=yes 
+  service:
+    enabled=yes
     name=docker
+  retries: 10
```
Comment (jordonsc) [inline ansible/roles/docker/tasks/main.yml]: wtf

**Review by jordonsc — CHANGES_REQUESTED** [review]

File: ansible/roles/docker/tasks/main.yml (hunk 2 above)
Comment (purinda) [reply ansible/roles/docker/tasks/main.yml]: yeah try for around 50secs, until the service becomes active. Feels like a hack looking back, I will try to use `notify` here.

File: src/os.sh (hunk 1 above)
Comment (purinda) [reply src/os.sh]: is there some doc which you can refer to for this sort of conventions.

Comment (purinda) [reply src/os.sh]: I will change it.

Comment (jordonsc) [reply src/os.sh]: the variable name matches the file name

Comment (purinda) [reply src/os.sh]: make sense, good ol' c style.

**Comment by purinda** [comment]
@jordonsc I fixed the review comments, have a look thanks. Also tested using Ubuntu AMIs.

**Review by jordonsc — APPROVED** [review]

**Pull request merged**